modules/javafx.web/src/test/java/test/javafx/scene/web/LocalStorageTest.java

```diff
@@ -0,0 +1,153 @@
+ * questions.
+ */
+
+package test.javafx.scene.web;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static java.lang.String.format;
+import java.io.RandomAccessFile;
+import java.nio.channels.FileLock;
+import java.util.concurrent.FutureTask;
+import java.io.File;
+import java.io.IOException;
+import java.util.Random;
+
+
+import javafx.scene.web.WebEngineShim;
+import javafx.concurrent.Worker;
+import javafx.scene.web.WebView;
+import javafx.scene.web.WebEngine;
+
+
+public class LocalStorageTest extends TestBase {
+
+    private static final File LOCAL_STORAGE_DIR = new File("LocalStorageDir");
+    private static final File PRE_LOCKED = new File("zoo_local_storage");
+
+    private static RandomAccessFile preLockedRaf;
+    private static FileLock preLockedLock;
+    private static final Random random = new Random();
+
+    private static void deleteRecursively(File file) throws IOException {
+        if (file.isDirectory()) {
+            for (File f : file.listFiles()) {
+                deleteRecursively(f);
+            }
+        }
+        if (!file.delete()) {
+            // If WebKit takes time to close the file, better
+            // delete it during VM shutdown.
+            file.deleteOnExit();
+        }
+    }
+
+    private WebEngine createWebEngine() {
+        return submit(() -> new WebEngine());
+    }
+
+    /* test localstorage instance */
+    void checkLocalStorageAfterWindowClose(WebEngine webEngine) {
+        load(new File("src/test/resources/test/html/localstorage.html"));
+        submit(() -> {
+            assertNotNull(webEngine.executeScript("localStorage;"));
+            getEngine().executeScript("window.close();");
+            assertNotNull(webEngine.executeScript("localStorage;"));
+        });
+    }
+
+    @BeforeClass
+    public static void beforeClass() throws IOException {
+        PRE_LOCKED.mkdirs();
+        File preLockedFile = new File(PRE_LOCKED, ".lock");
+        preLockedRaf = new RandomAccessFile(preLockedFile, "rw");
+        preLockedLock = preLockedRaf.getChannel().tryLock();
+        if (preLockedLock == null) {
+            fail(format("Directory [%s] is already locked "
+                    + "externally", PRE_LOCKED));
+        }
+    }
+
+    @AfterClass
+    public static void afterClass() throws IOException {
+        preLockedLock.release();
+        preLockedRaf.close();
+        deleteRecursively(LOCAL_STORAGE_DIR);
+        deleteRecursively(PRE_LOCKED);
+    }
+
+    @Test
+    public void testLocalStorage() throws Exception {
+        final WebEngine webEngine = getEngine();
+        webEngine.setJavaScriptEnabled(true);
+        webEngine.setUserDataDirectory(LOCAL_STORAGE_DIR);
```

Comment:
This should be done for all tests, not just this one. Remember that you shouldn't assume any particular order for the tests (tests are intended to be stateless).